03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,421 @@
+
+First let's save our clustered heatmap.
+
+### Save heatmap as a png
+
+You can easily switch this to save to a jpeg or tiff by changing the function 
+and file name within the function to the respective file suffix.
+
+```{r}
+# Open a png file
+png(file.path(
+  plots_dir,
+  "HeatmapSRP070849_nonannotated.png" # Replace file name with a relevant output plot name
+))
+
+# Print your heatmap
+pheatmap
+
+# Close the png file:
+dev.off()
+```
+
+Now, let's add some annotation bars to our heatmap.
+
+## Prepare metadata for annotation
+
+We are going to manipulate the metadata and add variables with the information that we would like to use to annotate the heatmap.
+
+```{r}
+# Let's prepare the annotation data.frame for the uncollapsed `DESeqData` set object which will be used to create the technical replicates heatmap
+annotation_df <- metadata %>%
+  # Create a variable to store the cancer type information
+  dplyr::mutate(
+    mutation = dplyr::case_when(
+      startsWith(refinebio_title, "TET2") ~ "TET2",
+      startsWith(refinebio_title, "IDH2") ~ "IDH2",
+      startsWith(refinebio_title, "WT") ~ "WT",
+      TRUE ~ "unknown"
+    )
+  ) %>%
+  # We want to select the variables that we want for annotating the technical replicates heatmap
+  dplyr::select(
+    refinebio_accession_code,
+    mutation,
+    refinebio_treatment
+  ) %>%
+  # The `pheatmap` function requires that the rownames of our annotation object matches the column names of our `DESeaDataSet` object
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### Create annotated heatmap
+
+You can create an annotated heatmap by providing our annotation object to the `annotation_col` argument of the `pheatmap` function.
+
+```{r}
+# Create and store the annotated heatmap object
+pheatmap_annotated <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE,
+    cluster_cols = TRUE,
+    show_rownames = FALSE,
+    annotation_col = annotation_df,
+    main = "Annotated Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
+```
+
+Now that we have annotation bars on our heatmap, we have a better idea of the sample variable groups that appear to cluster together.
+
+Let's save our annotated heatmap.
+
+### Save annotated heatmap as a png
+
+You can easily switch this to save to a jpeg or tiff by changing the function and file name within the function to the respective file suffix.
+
+```{r}
+# Open a png file
+png(file.path(
+  plots_dir,
+  "HeatmapSRP070849_annotated.png" # Replace file name with a relevant output plot name
+))
+
+# Print your heatmap
+pheatmap_annotated
+
+# Close the png file:
+dev.off()
+```
+
+# Further learning resources about this analysis
+
+- [Make heatmaps in R with pheatmap](https://slowkow.com/notes/pheatmap-tutorial/) [@make-heatmaps]
+
+For more complex heatmaps, you may want to take a look at:
```

Comment:
I like that you are including information to describe the resources but 1) Can you keep it in the same bullet point line? and 2) This description doesn't tell me much beyond what the title of `ComplexHeatmap` already told me 😄 .

Reply:
@cansavvy I _believe_ I fixed the description. Let me know if the new description is suffice or if you think users may require more context in the notebook before clicking the link.